client/src/PagesBody/ClassChoicePage/ClassChoicePage.js

```diff
@@ -37,6 +39,8 @@ const ClassChoicePage = () => {
           obedienceClassesObject.forEach((element) => {
             classes.push(element.classNumber);
           });
+
+          setIsPending(false)
```

Comment:
This should stop the "pending" state in all situations - whether the res.stataus was 200 or any other, perhaps move it outside of the if block.